app/src/main/java/org/oppia/android/app/settings/profile/ProfileListFragmentPresenter.kt

```diff
@@ -65,13 +69,13 @@ class ProfileListFragmentPresenter @Inject constructor(
   ) {
     binding.profile = profile
     binding.root.setOnClickListener {
-      activity.startActivity(
-        ProfileEditActivity.createProfileEditActivity(
-          activity,
-          profile.id.internalId,
-          isMultipane
-        )
-      )
+      if (isMultipane) {
+        val profileEditListener = (activity as LoadProfileEditListener)
+        profileEditListener.loadProfileEdit(profile.id.internalId, profile.name)
+      } else {
+        val routeToProfileEditListener = (activity as RouteToProfileEditListener)
```

Comment:
@BenHenning this line gives me a ClassCastException wheras the line 73 run completely fine. No idea here please help.

Reply:
@yash10019coder have you tried debugging what the activity was at the time of the exception occurring? I'm wondering if this fragment is being hosted somewhere other than AdministratorControlsActivity (which I think it is), so other activities may also need to be wired up for routing to work correctly.

Reply:
this is fixed .